src/dune_rules/lib.mli

```diff
@@ -148,6 +148,15 @@ module DB : sig
   val resolve_when_exists :
     t -> Loc.t * Lib_name.t -> lib Resolve.t option Memo.t
 
+  (** Like [resolve], but will return the first library in a list that is found
+      and raise an error if none of them can be found. *)
+  val resolve_first : t -> (Loc.t * Lib_name.t) list -> lib Resolve.Memo.t
```

Comment:
Don't make changes to this modules unless its strictly necessary please.

Reply:
Yes, I will revert this.